### DISCUSSION TITLE
Problematic source strings for translations

### DISCUSSION THREAD

**Discussion by Olf0** [sailfishos-patches/patchmanager#157] [Q&A] [answered]
This thread is an open discussion about problematic source strings for translations, i.e. in https://github.com/sailfishos-patches/patchmanager/blob/master/translations/settings-patchmanager.ts

Because some of the English language (source) strings are not well chosen, many translations either come up with awkward, literal translations or work around this by abstractly translating what the translator thinks the original (English) string was intended to mean (but hence are not really translating, rather guessing).

The proper solution in these cases is to enhance the source string, in order to alleviate the issue for all translators and the users of Patchmanager in English (presumably the majority).

Thus, if you as a translator believe a source string should be enhanced (with or without a suggestion how), please discuss it here or pose a pull request enhancing a [source string](https://github.com/sailfishos-patches/patchmanager/blob/master/translations/settings-patchmanager.ts).

Examples:
- https://github.com/sailfishos-patches/patchmanager/pull/155#issuecomment-961252475
- The term "engine" results in a couple of awkward translations (e.g. ["motorn" in SV](https://github.com/sailfishos-patches/patchmanager/blob/master/translations/settings-patchmanager.ts))

**Comment by Olf0** [comment, marked as answer]
Pondering about this resulted in [a glossary with some explanations ("Terms and wording" in the Wiki here](https://github.com/sailfishos-patches/patchmanager/wiki/Terms-and-wording)) and [a draft wiki page about the technical aspects of l10n / i18n for Patchmanager and Patches for it](https://github.com/sailfishos-patches/patchmanager/wiki/Translations-(%22i18n%22,-%22l10n%22)).

As always, constructive feedback addressing any aspect (from typos to logic etc.) of these documents, their content and the resulting choice of source strings (in English) is welcome.

**Comment by tuplasuhveli** [comment]
First of all, good job with the glossary!

I gathered some strings I found to be in need of improvement, although not all are related to translations. Everyone interested, please feel free comment and express your opinions.

1. Menu entries, when pressing and holding a patch:

- Patch details
- Activate
- Deactive
- Remove

In terms of consistency, should the first one be named as just "Details"?

2. "Openrepos" on the About page

Since OpenRepos is stylised with a capital 'R' on openrepos.net, should this practice be followed in Patchmanager as well?

3. "Version check: Strict / No check" on the Settings page

Would options "Enabled" and "Disabled" be more logical? Plus, I think this would look cleaner without the colon.

4. "Sources" and "Sources and Issue tracker" on the About page

Would it be a good idea to replace "sources" with "source code" to make it less ambiguous? 

5. "May conflict with:" on the About page

In most indoeuropean languages this doesn't cause any issues with translations, since prepositions is the most common way to express one thing's relation to another thing. However, in other languages, postpositions and use of cases are more common ways to express the same thing.

Therefore, I suggest modifying this string to something like this:

a.) "May conflict with the following Patches:"
b.) "This Patch may conflict with the following Patches:"

**Reply by nephros** [reply]
Thanks for the suggestions!

> 
>     3. "Version check: Strict / No check" on the Settings page
> 
> 
> Would options "Enabled" and "Disabled" be more logical? Plus, I think this would look cleaner without the colon.

Well the intention is to have other options in the future, i.e. being able to select Strict/Less strict/No checking. That's why the UI element is a dropdown, and the wording is as it is.  
There's no roadmap however when such a change might appear.

**Comment by Olf0** [comment]
@tuplasuhveli, WRT your points 1, 2, 4 and 5: **Yes**, I could not agree more to these suggestions!

Pull requests are welcome.
Mind that [altering text strings in the source code files also implies to update the source `.ts` file](https://github.com/storeman-developers/harbour-storeman/blob/devel/translations/README.md#updating-the-source-ts-file-with-source-strings-from-source-code) (i.e here, updating [`translations/settings-patchmanager.ts`](https://github.com/sailfishos-patches/patchmanager/blob/master/translations/settings-patchmanager.ts)).

**Reply by tuplasuhveli** [reply]
Thanks for the encouraging feedback @Olf0!

Once I'm done with today's work, I'll create the corresponding PRs.

**Reply by Olf0** [reply]
Thank you very much!

Side note: Now I notice that I used my usual phrase "Pull requests are welcome." and you responded with the plural I used.
Here, a single PR is fully sufficient IMO.

**Reply by Olf0** [reply]
Thanks, PR #493 is looking good; I intend go through it in detail, soon.

*Edit:* Done.

**Reply by Olf0** [reply]
This sub-thread was resolved by finishing and merging PR #493.
